src/ctelnet.cpp

```diff
@@ -964,26 +964,39 @@ QString cTelnet::getNewEnvironValueSystemType()
     QString systemType;
 
     // "SYSTEMTYPE" Inspired by https://www.rfc-editor.org/rfc/rfc1340.txt
-#if (defined(Q_OS_MAC) || defined(Q_OS_MACOS))
-    systemType = qsl("MACOS");
-#elif defined(Q_OS_WIN64)
-    systemType = qsl("WIN64");
+    // Ordering redone to follow general format of TLuaInterpreter::getOs()
+#if defined(Q_OS_CYGWIN)
+    // Try for this one before Q_OS_WIN32 as both are likely to be defined on
+    // a Cygwin platform
+    systemType = qsl("CYGWIN");
 #elif defined(Q_OS_WIN32)
+    // This is defined on BOTH Win32 and Win64 hosts - but it reflects
+    // the build machine rather than the run-time one and our published
+    // builds are actually 32-bit ones that can run on either. If we
+    // really wanted to distinguish between the two bit-nesses we'd have
+    // to do that at run-time - and we can probably leave off doing that
+    // until we officially publish 64 bit builds specifically for Win64
+    // machines:
     systemType = qsl("WIN32");
-#elif defined(Q_OS_BSD4)
-    nsystemType = qsl("BSD4");
```

Comment:
This is the reason the code wouldn't build for me. Since `Q_OS_BSD4` is defined for all three of the (Open>Net>Free)BSD OS's they would all FTBFS here...

Reply:
... as that check was mistakenly before the more OS specific ones which should have been before it.